src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java

```diff
@@ -346,14 +356,53 @@ protected ExtraButton createSubmitButton() {
      * @return
      */
     protected ExtraButton createCreateRequisitionButton() {
-        ExtraButton clearButton = new ExtraButton();
-        clearButton.setExtraButtonProperty("methodToCall.createRequisition");
-        clearButton.setExtraButtonSource("${" + KFSConstants.EXTERNALIZABLE_IMAGES_URL_KEY + "}buttonsmall_create_req.gif");
-        clearButton.setExtraButtonAltText("Create Req");
-        clearButton.setExtraButtonOnclick("window.open('" + ConfigContext.getCurrentContextConfig().getProperty(KFSConstants.APPLICATION_URL_KEY)
-                + "/purapRequisition.do?methodToCall=createReqFromIWantDoc&docId=" + getDocument().getDocumentNumber()
-                + "');return false;");
-        return clearButton;
+        ExtraButton createReqButton = new ExtraButton();
+        createReqButton.setExtraButtonProperty("methodToCall.createRequisition");
+        createReqButton.setExtraButtonSource("${" + KFSConstants.EXTERNALIZABLE_IMAGES_URL_KEY + "}buttonsmall_create_req.gif");
+        createReqButton.setExtraButtonAltText("Create Req");
+        if (getEditingMode().containsKey(CUPurapConstants.IWNT_DOC_EDIT_CONTRACT_INDICATOR) && isInOrgHierarchyNode(getDocument())) {
+            createReqButton.setExtraButtonOnclick(
+                    " if((document.getElementsByName('document.contractIndicator'))[1].checked) " 
+                            + " { " 
+                            + " if(confirm('" + getContractWarningMessage() + "')){ " 
+                            + "window.open('"
+                            + ConfigContext.getCurrentContextConfig().getProperty(KFSConstants.APPLICATION_URL_KEY)
+                            + "/purapRequisition.do?methodToCall=createReqFromIWantDoc&docId="
+                            + getDocument().getDocumentNumber() 
+                            + "'); return false;" 
+                            + " } " 
+                            + " else { " 
+                            + "return false; "
+                            + " } } "
+                            + " else {"
+                            + "window.open('" 
+                            + ConfigContext.getCurrentContextConfig().getProperty(KFSConstants.APPLICATION_URL_KEY)
+                            + "/purapRequisition.do?methodToCall=createReqFromIWantDoc&docId=" 
+                            + getDocument().getDocumentNumber()
+                            + "'); return false; "
+                            + " } "
+                    );
+        } else if (!getEditingMode().containsKey(CUPurapConstants.IWNT_DOC_EDIT_CONTRACT_INDICATOR)
+                        && isInOrgHierarchyNode(getDocument()) && KFSConstants.ParameterValues.YES.equalsIgnoreCase(getIWantDocument().getContractIndicator())) {
```

Comment:
The other areas of this PR's code are using `KRADConstants.YES_INDICATOR_VALUE` for the "Y" comparisons. Please update this spot to use that KRAD constant for consistency.

Reply:
updated